core/src/test/resources/sql/planner.iq

```diff
@@ -351,4 +351,64 @@ FROM "scott"."EMP";
 
 !ok
 
+# [CALCITE-4617] Wrong offset when SortJoinTransposeRule pushes a Sort with an offset
```

Comment:
Could we also have an example of a query where sort/offset/fetch is actually pushed inside the join? (and confirm the results match with vs without the rule)
I think the original PR has a [couple of examples](https://github.com/apache/calcite/pull/2417/files#diff-feaf84fbc8f315359bbce262f7d336969cfae0c1ade39e0dd31db197b9bec27d):

Reply:
I have added these test cases, please help review them again. @rubenada